### PR TITLE
I736 rm validation classes

### PR DESCRIPTION
**Pull request by laneb**
Addressing #736.

### CODE REVIEW

File: src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala [hunk 1]
```diff
@@ -93,7 +97,7 @@ object SmvDQM {
   def apply() = new SmvDQM()
 }
 
-class DQMValidator(dqm: SmvDQM) extends ValidationTask {
+class DQMValidator(dqm: SmvDQM, isPersist: Boolean) {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala]: Please add a comment to describe the class and what the parameters are.

File: src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala [hunk 2]
```diff
@@ -148,10 +152,72 @@ class DQMValidator(dqm: SmvDQM) extends ValidationTask {
...
+  }
+
+  private def readPersistsedValidationFile(path: String): Try[DqmValidationResult] = {
+    Try({
+      val json = SmvReportIO.readReport(path)
+      DqmValidationResult(json)
+    })
+  }
+
+  def validate(df: DataFrame, hadAction: Boolean, path: String = "") = {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala]: add function document as the `hadAction` and `path` here are not super obvious.

File: src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala [hunk 3]
```diff
@@ -148,10 +152,72 @@ class DQMValidator(dqm: SmvDQM) extends ValidationTask {
...
+        }.get
+      } else {
+        runValidation(df, forceAction, path)
+      }
+
+      terminateAtError(result)
+      result
+  }
+
+  def runValidation(df: DataFrame, forceAction: Boolean, path: String = ""): DqmValidationResult = {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/dqm/SmvDQM.scala]: yup, another doc needed.

**Comment by laneb** [comment]
Requested comments have been added.

**Pull request merged**